packages/cli/src/api/formats/po.js

```diff
@@ -1,104 +1,93 @@
 // @flow
 import fs from "fs"
+import { dirname } from "path"
 import * as R from "ramda"
+import { format as formateDate } from "date-fns"
 
-import gettextParser from "gettext-parser"
-import { prettyOrigin } from "../utils"
+import PO from "pofile"
+import { joinOrigin, splitOrigin } from "../utils"
 import type { TranslationsFormat } from "../types"
 
-const pad2 = num => ("00" + num.toString()).slice(-2)
-
-const getHeaders = ({ language = "no", ...headers } = {}) => {
-  const date = new Date()
-  const today = `${date.getFullYear()}-${pad2(date.getMonth() + 1)}-${pad2(
-    date.getDate()
-  )}`
-
-  return {
-    "project-id-version": "Project Name",
-    "pot-creation-date": today,
-    "po-revision-date": today,
-    "last-translator": "",
-    "language-team": "",
-    "mime-version": "1.0",
-    "content-type": "text/plain; charset=utf-8",
-    "content-transfer-encoding": "8bit",
-    language,
-    "x-generator": "@lingui/cli",
-    ...headers
-  }
-}
-
-function writeHeaders(headers) {
-  const res = []
-  res.push(`msgid ""`)
-  res.push(`msgstr ""`)
-
-  Object.keys(headers).forEach(key => res.push(`"${key}: ${headers[key]}\\n"`))
-
-  return res
-}
+const getCreateHeaders = language => ({
+  "Project-Id-Version": dirname(process.cwd()),
+  "POT-Creation-Date": formateDate(new Date(), "YYYY-MM-DD HH:mmZZ"),
+  "Mime-Version": "1.0",
+  "Content-Type": "text/plain; charset=utf-8",
+  "Content-Transfer-Encoding": "8bit",
+  "X-Generator": "@lingui/cli",
+  Language: language
+})
 
 const serialize = R.compose(
   R.values,
   R.mapObjIndexed((message, key) => {
-    const translation = message.translation || ""
-    const escape = str => str.toString().replace(/"/g, '\\"')
-
-    return [
-      "",
-      message.description && `#. ${message.description}`,
-      ...(message.origin
-        ? // print one origin per line (don't join multiple origins with comma)
-          message.origin.map(origin => `#: ${prettyOrigin([origin])}`)
-        : []),
-      message.obsolete && `#, obsolete`,
-      `msgid "${escape(key)}"`,
-      `msgstr "${escape(translation)}"`
-    ]
-      .filter(item => typeof item === "string")
-      .join("\n")
+    const item = new PO.Item()
+    item.msgid = key
+    item.msgstr = message.translation
+    item.comments = message.comments || []
+    item.extractedComments = message.description ? [message.description] : []
+    item.references = message.origin ? message.origin.map(joinOrigin) : []
+    item.obsolete = message.obsolete
+    return item
   })
 )
 
-function parseOrigin(origins) {
-  if (!origins) return
+const getMessageKey = R.prop("msgid")
+const getTranslations = R.prop("msgstr")
+const getExtractedComments = R.prop("extractedComments")
+const getTranslatorComments = R.prop("comments")
+const getOrigins = R.prop("references")
+const isObsolete = R.either(R.path(["flags", "obsolete"]), R.prop("obsolete"))
+
+const deserialize = R.map(item => ({
```

Comment:
I am wondering if there is some better way with Ramda so the `item` does not need to be passed in every line. Closest I could find is `R.evolve`, but that cannot do a transformation of keys, just values.

Reply:
I think you could use [`R.applySpec`](https://ramdajs.com/docs/#applySpec) instead of `R.evolve`.

Kudos for thinking functionally! 👍

Reply:
Alright, I missed that one. It actually works, the only problem is with comments which I am not sure how to write to extract both types of comments and concat it together. I am not sure how to tell it to create a function that accepts `item` and passes it only to those two functions within `concat`.

```
  R.applySpec({
    comments: R.concat(
      getTranslatorComments,
      R.compose(
        R.tail,
        getExtractedComments
      )
    ),
```

Ramda is something I want to tackle eventually, but it's somewhat hard :)

Reply:
```js
R.compose(
  R.apply(R.concat),
  R.juxt([
    getTranslatorComments,
    R.compose(R.tail, getExtractedComments)
  ])
)
```

See [repl](https://ramdajs.com/repl/?v=0.25.0#?function%20getTranslatorComments%28item%29%20%7B%0A%20%20return%20item.comments%0A%7D%0A%0Afunction%20getExtractedComments%28item%29%20%7B%0A%20%20return%20item.description%0A%7D%0A%0Avar%20deserialize%20%3D%20R.compose%28%0A%20%20R.apply%28R.concat%29%2C%0A%20%20R.juxt%28%5B%0A%20%20%20%20getTranslatorComments%2C%0A%20%20%20%20R.compose%28R.tail%2C%20getExtractedComments%29%0A%20%20%5D%29%0A%29%0A%0Adeserialize%28%7B%0A%20%20comments%3A%20%27comments%27%2C%0A%20%20description%3A%20%27description%27%0A%7D%29%0A)

Although I'm not sure if like it over:

```js
item => R.concat(
    getTranslatorComments(item),
    R.tail(getExtractedComments(item))
)
```

JavaScript is missing functional operators to make this readable...

Reply:
Ah lol, I am so much thinking in Ramda terms that I did not realize I can make such simple function there :D Yea, in this case, it will be definitely more readable. Thanks!

Reply:
Alright, it's more functional now, but I am not that much convinced it's prettier :) Could be interesting to measure if it's actually faster to create such a big curried function opposed to creating it on fly.